api/client.go

```diff
@@ -260,37 +262,25 @@ func (c *client) WorkspaceUsers(p WorkspaceUsersParam) (users []dto.User, err er
 		return users, err
 	}
 
-	r, err := c.NewRequest(
+	err = c.paginate(
 		"GET",
 		fmt.Sprintf("v1/workspaces/%s/users", p.Workspace),
-		nil,
-	)
-	if err != nil {
-		return users, err
-	}
-
-	_, err = c.Do(r, &users, "WorkspaceUsers")
-	if err != nil {
-		return users, err
-	}
-
-	if p.Email == "" {
-		return users, nil
-	}
-
-	uCopy := []dto.User{}
-	for i := 0; i < len(users); i++ {
-		if !strings.Contains(
-			strings.ToLower(users[i].Email),
-			strings.ToLower(p.Email),
-		) {
-			continue
-		}
+		PaginationParam{AllPages: true},
```

Comment:
you need to change this too
```suggestion
		p.PaginationParam,
```

Reply:
I had that at first, but then it doesn't list all pages?

Reply:
you need to change the cmd who calls this function.

the api/* is used as a SDK for clockify for other stuff... so the callers must set the pagination option.

like here:
https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/pkg/cmd/tag/tag.go#L93

these are the files that use the function:
- https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/pkg/search/user.go#L19
- https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/pkg/cmdcomplutil/user.go#L27
- https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/pkg/cmd/user/user.go#L58
- https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/pkg/cmd/config/init/init.go#L69
- https://github.com/lucassabreu/clockify-cli/blob/97dedc8cc70c6dd8c6dd7a5e224dcb2efcd84801/api/client.go#L755

Reply:
@lucassabreu I think I fixed it? I moved the `AllPages` setting to the call-sites so that the function`WorkspaceUsers` itself is generic.

Reply:
Thanks for pointing me in the right direction!